src/main/java/com/conveyal/r5/profile/FrequencyRandomOffsets.java

```diff
@@ -179,10 +201,20 @@ public void randomize () {
                     }
                 }
             }
-
             if (remainingAfterPreviousRound == remaining && remaining > 0) {
                 throw new IllegalArgumentException("Cannot solve phasing, you may have a circular reference!");
             }
+            // Copy results of randomization to a Map keyed on TripSchedules (instead of TripPattern index ints)
+            offsetsForTripSchedule.clear();
+            for (TIntObjectIterator<int[][]> it = offsets.iterator(); it.hasNext(); ) {
+                it.advance();
+                TripPattern tripPattern = data.tripPatterns.get(it.key());
+                int[][] offsetsForTrip = it.value();
+                for (int ts = 0; ts < tripPattern.tripSchedules.size(); ts++) {
+                    TripSchedule tripSchedule = tripPattern.tripSchedules.get(ts);
+                    offsetsForTripSchedule.put(tripSchedule, offsetsForTrip[ts]);
```

Comment:
Looking at this again, I think `offsetsForTrip[ts]` can be null when it has no frequency patterns and there's no point in storing those.